### DISCUSSION TITLE
Definition inside a scrollable table.

### DISCUSSION THREAD

**Discussion by dajones537** [uswds/uswds#5745] [General]
My app has a lot of tables, and some tooltips (definitions in the USWDS terminology). If I put a tooltip in a column header, it works, other things being equal. However, some of my tables are scrollable and have a lot of columns. In these scenarios, the tooltip appears to always exist at its original X value. Example:

````
<div class="usa-table-container--scrollable" tabindex="0">
  <table class="usa-table usa-table--scrollable">
    <caption>Scrollable table</caption>
    <thead>
      <tr>
        <th scope="col">Document title</th>
        <th scope="col">
            <span class="definition js-definition">
              <button type="button" class="definition__trigger js-definition__trigger">
                Acid Rain
              </button>
              <span class="definition__tooltip js-definition__tooltip" role="tooltip">
                <dfn class="definition__term">Acid Rain</dfn>
                The result of sulfur dioxide (SO<sub>2</sub>) and nitrogen oxides (NO<sub>x</sub>) reacting in the atmosphere with water and returning to earth as rain, fog, or snow. Broadly used to include both wet and dry deposition. The <a href="http://www.epa.gov/acidrain/index.html">acid rain page</a> provides a great deal of information about this issue.
              </span>
            </span>
        </th>
        <th scope="col">Year</th>
        <th scope="col">Year</th>
        <th scope="col">Year</th>
        <th scope="col">Year</th>
        <th scope="col">Year</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th scope="row">Declaration of Independence</th>
        <td>Statement adopted by the Continental Congress declaring independence from the British Empire.</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
      </tr>
      <tr>
        <th scope="row">Bill of Rights</th>
        <td>The first ten amendments of the U.S. Constitution guaranteeing rights and freedoms.</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
      </tr>
      <tr>
        <th scope="row">Declaration of Sentiments</th>
        <td>A document written during the Seneca Falls Convention outlining the rights that American women should be entitled to as citizens.</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
      </tr>
      <tr>
        <th scope="row">Emancipation Proclamation</th>
        <td>An executive order granting freedom to slaves in designated southern states.</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
      </tr>
    </tbody>
  </table>
````
Here is how this looks at the outset:
![image](https://github.com/uswds/uswds/assets/123194829/ad7675e1-0dca-4d7b-91eb-d5643c7b87e8)

But after scrolling, we can see that the tooltip is now in the wrong place:
![image](https://github.com/uswds/uswds/assets/123194829/4e2f6e33-eeda-4802-9ece-8770fc1362b6)

I wanted to know if anyone else has had success with tooltips inside of scrollable tables. Thanks.

**Comment by amyleadem** [comment]
Hello @dajones537,
Thanks for reaching out. I tested this out using the standard implementation found on the [Tooltip component page](https://designsystem.digital.gov/components/tooltip/), and I see that the standard implementation does follow along with its parent element after horizontal scroll.

I noticed your code follows a different structure, I'm guessing because it has longer content and some rich HTML.  Would it fit in your project to follow the standard code pattern instead? 

Based on the [guidance on the tooltip page](https://designsystem.digital.gov/components/tooltip/#when-to-consider-something-else-tooltip), I also wonder if another component might be a better fit for this more complex content:

> Lengthy descriptions. Tooltips are microcopy, and should be brief. Don’t use a tooltip if you need a lot of text.[^1]

[^1]: https://designsystem.digital.gov/components/tooltip/#when-to-consider-something-else-tooltip

Here is what I found when I used the standard implementation:

**Before horizontal scroll:**

![image](https://github.com/uswds/uswds/assets/93996430/873c63a1-935d-4036-be36-5b497a5ce193)

**After horizontal scroll:**

![image](https://github.com/uswds/uswds/assets/93996430/43e77476-7dbe-4468-8c8f-0b07c25778dd)


Here is the code I used to get these results:

```html
<div class="usa-table-container--scrollable" tabindex="0">
  <table class="usa-table">
    <caption>
      Scrollable table
    </caption>
    <thead>
      <tr>
        <th scope="col">Document title</th>
        <th scope="col">
          <span class="usa-tooltip" data-position="bottom" title="Information about acid rain">
            Acid Rain
          </span>
        </th>
        <th scope="col">Year</th>
        <th scope="col">Year</th>
        <th scope="col">Year</th>
        <th scope="col">Year</th>
        <th scope="col">Year</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th scope="row">Declaration of Independence</th>
        <td>Statement adopted by the Continental Congress declaring independence from the British Empire.</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
      </tr>
      <tr>
        <th scope="row">Bill of Rights</th>
        <td>The first ten amendments of the U.S. Constitution guaranteeing rights and freedoms.</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
      </tr>
      <tr>
        <th scope="row">Declaration of Sentiments</th>
        <td>A document written during the Seneca Falls Convention outlining the rights that American women should be entitled to as citizens.</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
      </tr>
      <tr>
        <th scope="row">Emancipation Proclamation</th>
        <td>An executive order granting freedom to slaves in designated southern states.</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
        <td>1776</td>
      </tr>
    </tbody>
  </table>
</div>
<p>* in billions of dollars. Data for illustration purposes only.</p>
```

Hoping that helps. Let us know if you have any questions!

**Reply by dajones537** [reply]
Thanks for the response. I realized I shouldn't have used the term tooltip. I didn't realize USWDS actually has something called tooltip, and I was actually using the term tooltip to refer to definitions, like this: https://www.epa.gov/themes/epa_theme/pattern-lab/?p=viewall-components-definition. So while tooltips may work, it's actually definitions I'm talking about, and scrolling does break definitions for me.
